src/custom/state/swap/hooks.tsx

```diff
@@ -347,7 +367,7 @@ export function useDerivedSwapInfo(): DerivedSwapInfo {
         currencyBalances,
         parsedAmount,
         inputError,
-        v2Trade: v2Trade ?? undefined, // mod
+        v2Trade: (!isEthFlowBuyOrder && v2Trade) || undefined, // mod
```

Comment:
Not very clean, open to suggestions.

I'm not returning the trade object if this is an ethflow buy order

Reply:
Would be great to add an explanation, why don't you return the trade?

Reply:
Right, forgot to add that, sorry!

Not returning the trade obj removes all the order fields in the form, IMO making it clearer the trade is not supported

See the comparison:

| sell order | buy order |
|-|-|
| ![Screen Shot 2022-11-25 at 08 32 53](https://user-images.githubusercontent.com/43217/203936670-2592bc75-ce04-4186-9f76-ecd9f7bdc8d6.png) | ![Screen Shot 2022-11-25 at 08 33 09](https://user-images.githubusercontent.com/43217/203936685-997c1488-9270-4938-9f01-09bc69c1db64.png) |

Of course one could go to each component and re-use the logic to detect ethflow buy order (`!isExactIn && isEthFlow`) but it's the same rule as when there's no `trade` obj and because I'm lazy I find this easier =P
Also less places to spread the logic around.

Reply:
> Would be great to add an explanation, why don't you return the trade?

I guess you also mean in the code (to add a small comment?)